File: two_sum-II-input_arr_is_sorted_167.py
from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:

        left, right = 0, len(numbers) - 1

        while left <= right:
            cur_sum = numbers[left] + numbers[right]
            if cur_sum == target:
                return [left+1, right+1]
            elif cur_sum > target:
                right -= 1
            else:
                left += 1


s = Solution()
s.twoSum([5, 25, 75], 100)

File: two_sum-II-input_arr_is_sorted_167.js
function twoSum(numbers, target) {
  let left = 0;
  let right = numbers.length - 1;

  while (left < right) {
    const sum = numbers[left] + numbers[right];
    if (sum < target) {
      left += 1;
    } else if (sum > target) {
      right -= 1;
    } else {
      return [left, right];
    }
  }
}
